### PR TITLE
CircleCI preview environment docs

### PR DESCRIPTION
**Pull request by ryan-cahill**
## Purpose
Closes https://gitlab.com/architect-io/product/-/issues/285

## Changes
* Adds docs for CircleCI-based preview environments

### CODE REVIEW

File: deployments/automated-previews.mdx [hunk 1]
```diff
@@ -274,3 +274,161 @@ pipelines:
...
+</Note>
+
+<Note>
+  `ARCHITECT_PASSWORD` must be a [personal access
+  token](/deployments/personal-access-tokens).
+</Note>
+
+### Preview environment creation trigger
+
+The file `circleci-create-preview.yml` will trigger a CircleCI workflow to be run when a pull request is created or updated in GitHub. Note that `GHA_Meta` is set to `ARC_CREATE_PREVIEW`
```
Comment (Zach-at-Arch) [inline deployments/automated-previews.mdx:296]: Does the user need to set `ARC_CREATE_PREVIEW` to anything?

**Comment by Zach-at-Arch** [comment]
@ryan-cahill everything looks good, just had one clarifying question - otherwise, it's got my approval

File: deployments/automated-previews.mdx (hunk 1 above)
Comment (ryan-cahill) [reply deployments/automated-previews.mdx:296]: No, that's just set so that CircleCI knows which job to run

**Review by Zach-at-Arch — APPROVED** [review]

**Comment by tjhiggins** [comment]
@davidthor fyi

**Pull request merged**